keepassxc-browser/keepassxc-browser.js

```diff
@@ -1126,7 +1126,32 @@ cipFields.useDefinedCredentialFields = function() {
     return false;
 };
 
+MutationObserver = window.MutationObserver || window.WebKitMutationObserver;
+
+// Detects DOM changes in the document
+let observer = new MutationObserver(function(mutations, observer) {
+    if (document.visibilityState !== 'hidden') {

```

Comment:
I would prefer you restructure this whole callback function with return guards. In JS it's really tempting to create deeply nested structures, which are hard to understand.